### DISCUSSION TITLE
完善跨域配置

### DISCUSSION THREAD

**Discussion by SSSSSSSan** [xfgryujk/blivechat#192] [想法]
目前仅靠将配置文件中的tornado_xheaders设置为true不接受跨域websocket，导致我这种用api网关做反代的抽象无法启用通过服务器转发功能
要改的不多，，配置文件加上
```
# 允许跨域的域名，使用逗号分隔
# allowed_origins = https://123.com,http://678.com
allowed_origins = 
```
在config.py的AppConfig._load_app_config中加入
```
# 读取允许跨域的域名，并按逗号分割成列表  
allowed_origins_str = app_section.get('allowed_origins', '')  
self.allowed_origins = [origin.strip() for origin in allowed_origins_str.split(',')]
```
再将chat.py中的check_origin修改为
```
 def check_origin(self, origin):
     cfg = config.get_config()
     if cfg.debug:
         return True
     if origin in cfg.allowed_origins:
         return True  
     return super().check_origin(origin)
```

**Comment by xfgryujk** [comment]
为什么网关会改域名，配了什么奇怪的重写吗

**Reply by SSSSSSSan** [reply]
额是这样
首先 我有一个域名指向api网关
api网关和后端不在同一个网络，用另外的二级域名指向了后端，同时端口也改写了

然后websocket是透传，所以网关的host就被透传进来了，就对不上了

嗯 挺抽象

**Reply by xfgryujk** [reply]
> 额是这样 首先 我有一个域名指向api网关 api网关和后端不在同一个网络，用另外的二级域名指向了后端，同时端口也改写了
> 
> 然后websocket是透传，所以网关的host就被透传进来了，就对不上了
> 
> 嗯 挺抽象

小众的需求你自己改吧
